src/interpreter/interpreter.rs

```diff
@@ -4,22 +4,438 @@ use crate::ir::ast::Expression;
 use crate::ir::ast::Name;
 use crate::ir::ast::Statement;
 
-type IntValue = i32;
 type ErrorMessage = String;
 
-type Environment = HashMap<Name, i32>;
+#[derive(Debug, Clone, PartialEq)]
+pub enum EnvValue {
+    CInt(i32),
+    CReal(f32),
```

Comment:
For consistency with (most) other languages and standards, maybe this can be called CFloat ,CFloating or something like that? If we are to be strict, floating point numbers are not real numbers, as real numbers have infinite precision.